OrbitCore/LinuxEventTracer.h

```diff
@@ -5,46 +5,103 @@
 //-----------------------------------
 // Author: Florian Kuebler
 //-----------------------------------
-#pragma once
+
+#ifndef ORBIT_CORE_LINUX_EVENT_TRACER_H_
+#define ORBIT_CORE_LINUX_EVENT_TRACER_H_
 
 #include <atomic>
 #include <functional>
 #include <thread>
+#include <utility>
 
 #include "CallstackTypes.h"
 #include "LibunwindstackUnwinder.h"
+#include "OrbitFunction.h"
+#include "PrintVar.h"
+
+class LinuxEventTracerThread {
+ public:
+  LinuxEventTracerThread(pid_t pid, double frequency,
+                         std::vector<Function*> functions)
+      : pid_(pid), frequency_(frequency), functions_(std::move(functions)) {}
+
+  LinuxEventTracerThread(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread& operator=(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread(LinuxEventTracerThread&&) = default;
+  LinuxEventTracerThread& operator=(LinuxEventTracerThread&&) = default;
+
+  void Run(const std::shared_ptr<std::atomic<bool>>& a_ExitRequested);
+
+ private:
+  static constexpr int32_t ROUND_ROBIN_BATCH_SIZE = 5;
+
+  pid_t pid_;
+  double frequency_;
+  std::vector<Function*> functions_;
+
+  uint64_t period_ns_ = 1000000;
+  int32_t cpus_ = 0;
+
+  bool ComputeFrequency();
+  void LoadNumCpus();
+  void HandleCallstack(pid_t tid, uint64_t timestamp,
```

Comment:
Does this have to be a member of this class?

Reply:
See the other comment.